packages/flutter_tools/lib/executable.dart

```diff
@@ -124,69 +133,85 @@ Future<Null> main(List<String> args) async {
     // Initialize the system locale.
     await intl.findSystemLocale();
 
-    return Chain.capture<Future<Null>>(() async {
+    Completer<int> runCompleter = new Completer<int>();
+    Chain.capture<Future<Null>>(() async {
```

Comment:
The exit code is reported via the completer (because we also need to reports in the error case), so this is still just `Future<Null>`.